_spinnaker_install_admin_guides/halyard-gitops.md

```diff
@@ -0,0 +1,180 @@
+
+As a pre-requisite, you should [store secrets outside of your configuration](../secrets-s3).
+
+## Configure Halyard
+In `halyard.yml`, configure the deployments you'd like to manage:
+
+```yaml
+halyard:
+  halconfig:
+    storage:
+      github:
+        # You can use encrypted secrets here to pass sensitive information
+        token: encrypted:...
+    configs:
+    - name: staging
+      github:
+        path: github.com:acmecorp/spinnaker-configs/staging
+        branch: master
+    - name: prod
+      github:
+        path: github.com:acmecorp/spinnaker-configs/prod
+        branch: master
+...
+```
+
+The `path` should be to the root of the directory containing the `config` file.
+
+Note: When using mutiple deployments in a single file or if the deployment name is not `default`, you should distinguish between deployments by using `deploymentName`:
+
+```yaml
+halyard:
+  halconfig:
+    storage:
+      github:
+        token: encrypted:...
+    configs:
+    - name: staging
+      github:
+        path: github.com:acmecorp/spinnaker-configs
+        branch: master
+        deploymentName: staging
+    - name: prod
+      github:
+        path: github.com:acmecorp/spinnaker-configs
+        branch: master
+        deploymentName: prod
+...
+```
+
+
+## Deploy Halyard as a service
+
+We recommend deploying Halyard in the same cluster as Spinnaker so validations are more accurate. You can however deploy Halyard in its own namespace.
+
+Create a config map for Halyard's configuration:
+```
+kubeconfig -n YOUR_NAMESPACE create configmap halyard-config --from-file=./halyard.yml
+```
+
+Then make a deployment manifest:
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  annotations:
+  labels:
+    app: halyard
+  name: halyard-armory
+  namespace: YOUR_NAMESPACE
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: halyard
+  template:
+    metadata:
+      labels:
+        app: halyard
+    spec:
+      containers:
+      - name: halyard
+        image: docker.io/armory/halyard-armory:{{ site.data.versions.halyard-armory-version }}
+        volumeMounts:
+        - name: halyard-config-volume
+          mountPath: /opt/spinnaker/config
+        ports:
+        - containerPort: 8064
+          protocol: TCP
+      volumes:
+      - name: halyard-config-volume
+        configMap:
+          name: halyard-config
+```
+
+We can then expose Halyard. Refer to [Kubernetes](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#expose) or your cloud provider documentation for more options:
+```
+kubectl -n YOUR_NAMESPACE expose deployment halyard-armory --type=LoadBalancer --port=80
+```
+
+Don't forget to:
+- Refer to your cloud provider documentation to create a secure endpoint.
+- Secure Halyard's endpoints by limiting IP sources or using security groups of your cloud provider.
+- Add a DNS record for Halyard
+
+# Deploy Spinnaker automatically
```

Comment:
It would be nice if there was a 4th option (local halyard) outlined here, juuuuuuuuuuust incase.

Reply:
That's the point of the doc (keep things simple) and based on the feedback see if we need to work on the remote hal option. Note that local hal is kind of irrelevant for (automated) gitops